client/chat/route.tsx

```diff
@@ -1,30 +1,59 @@
-import React from 'react'
-import { useRoute } from 'wouter'
-import { makeSbChannelId } from '../../common/chat'
+import React, { Suspense } from 'react'
+import { Route, RouteProps, Switch } from 'wouter'
+import { makeSbChannelId, SbChannelId } from '../../common/chat'
+import { hasAnyPermission } from '../admin/admin-permissions'
 import { replace } from '../navigation/routing'
+import { LoadingDotsArea } from '../progress/dots'
+import { useAppSelector } from '../redux-hooks'
 import { ConnectedChatChannel } from './channel'
+import { ChannelList } from './channel-list'
+import { CreateChannel } from './create-channel'
 
-export function ChannelRouteComponent(props: { params: any }) {
-  const [matches, params] = useRoute('/chat/:channelId/:channelName')
+const getLoadableChatAdminComponent = (isAdmin: boolean) =>
+  isAdmin
+    ? React.lazy(async () => ({
+        default: (await import('./admin')).ChatAdmin,
+      }))
+    : // TODO(2Pac): Create a better-designed, general component for this
```

Comment:
Can we stop sprinkling TODOs everywhere? It is just about as many characters to write this TODO as it is to write a component that translates this value to some common string, so please just do that if that's what you want. I can't imagine what else this is about, as this is just a span with text in it.